command/agent/auth/cert/cert.go

```diff
@@ -0,0 +1,65 @@
+package cert
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	hclog "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/command/agent/auth"
+)
+
+type certMethod struct {
+	logger    hclog.Logger
+	mountPath string
+	name      string
+}
+
+func NewCertAuthMethod(conf *auth.AuthConfig) (auth.AuthMethod, error) {
+	if conf == nil {
+		return nil, errors.New("empty config")
+	}
+
+	// Not concerned if the conf.Config is empty as the 'name'
+	// parameter is optional when using TLS Auth
+
+	c := &certMethod{
+		logger:    conf.Logger,
+		mountPath: conf.MountPath,
+		name:      "",
+	}
+
+	if conf.Config != nil {
+		nameRaw, ok := conf.Config["name"]
+		if !ok {
+			nameRaw = ""
```

Comment:
This is a bit non-idiomatic; just do `if ok` then do the conversion and setting in the if block.